youtube_dl/extractor/panopto.py

```diff
@@ -0,0 +1,277 @@
+                'uploader': 'Panopto Support',
+                'timestamp': 1449409251.8579998,
+            },
+            'playlist': [
+                {
+                    'md5': 'e22b5a284789ba2681e4fe215352d816',
+                    'info_dict': {
+                        'id': '15ad06ef-3f7d-4074-aa4a-87c41dd18f9c',
+                        'ext': 'mp4',
+                        'title': 'OBJECT',
+                    },
+                },
+                {
+                    'md5': 'a483b8116abbb04a7112a9a3ccc835ce',
+                    'info_dict': {
+                        'id': '7668d6b2-dc81-421d-9853-20653689e2e8',
+                        'ext': 'mp4',
+                        'title': 'DV',
+                    },
+                },
+            ],
+            'playlist_count': 2,
+        },
+    ]
+
+    @staticmethod
+    def _get_contribs_str(contribs):
+        """Returns a comma-delimited string of contributors."""
+        s = ''
+        for c in contribs:
+            display_name = c.get('DisplayName')
+            if display_name is not None:
+                s += '{0}, '.format(display_name)
+        return s[:-2] if len(contribs) else ''
+
+    def _real_extract(self, url):
+        """Extracts the video and stream information for the given Panopto hosted URL."""
+        video_id = self._match_id(url)
+        org = self._match_organization(url)
+
+        delivery_info = self._download_json(
+            'https://{0}.hosted.panopto.com/Panopto/Pages/Viewer/DeliveryInfo.aspx'.format(org),
+            video_id,
+            query={
+                'deliveryId': video_id,
+                'invocationId': '',
+                'isLiveNotes': 'false',
+                'refreshAuthCookie': 'true',
+                'isActiveBroadcast': 'false',
+                'isEditing': 'false',
+                'isKollectiveAgentInstalled': 'false',
+                'isEmbed': 'false',
+                'responseType': 'json',
+            }
+        )
+
+        if 'ErrorCode' in delivery_info:
+            self._downloader.report_warning("If the video you are trying to download requires you to sign-in, you will "
+                                            "need to provide a cookies file that allows the downloader to authenticate "
+                                            "with Panopto. If the error below is about unauthorized access, this is "
+                                            "most likely the issue.")
+            raise ExtractorError(
+                'API error: ({0}) {1}'.format(delivery_info.get('ErrorCode', '?'), delivery_info.get('ErrorMessage', '?'))
+            )
+
+        streams = []
+        for this_stream in delivery_info['Delivery']['Streams']:
+            new_stream = {
+                'id': this_stream['PublicID'],
+                'title': this_stream['Tag'],
+                'formats': [],
+            }
+            if 'StreamHttpUrl' in this_stream:
```

Comment:
```suggestion
            if ('StreamHttpUrl' in this_stream and this_stream['StreamHttpUrl'] is not None):

```
It looks like this now can be null/none